learn/framework/architecture.md

```diff
@@ -0,0 +1,66 @@
+---
+
+# Framework Architecture {: #framework-architecture } 
+
+## Introduction {: #introduction }
+
+Substrate is a software development kit (SDK) for building blockchains. This framework is the foundation and engine powering the Polkadot's relay chain, the parachains (such as the Tanssi network itself), and the ContainerChains deployed through Tanssi. 
+
+Written in the Rust Language and designed with a modular architecture, Substrate is extremely performant, flexible, and highly customizable, making it the best choice for developing blockchains.
+
+In this article, the architecture of a Substrate node is covered.
+
+## Architecture {: #architecture }
+
+The Substrate framework is designed for maximum customizability, providing a fully functional implementation for every important internal aspect of a blockchain. It allows developers to focus on the specifics of the use case and the runtime characteristics and to change any of the default features (should the need arise).
+
+The architecture of a Substrate node contains two main components:
+
+- **Core Client** - handles the communication with the outer world (other nodes, DApps, etc.), and many other internal responsibilities, such as storage and communication
+- **Runtime** - implements the custom logic of the Appchain, executes transactions, and manages the state transitions
+
+The end users can interact with the Appchain using the DApps (or directly via the node RPC endpoints) and get data or send transactions, which will remain queued until execution in the runtime.
+
+![Basic substrate node architecture](/images/learn/framework/architecture/architecture-1.png)
+
+## The Core Client {: #core-client }
+
+The core client comprises components responsible for everything in the operation of a node in the network except for what happens in the runtime. 
+
+Some of the main components are: 
+
+- **Networking** - this component handles the communication with the peers in the network (synchronizing blocks, propagating transactions, and so on) and exposes the endpoints that allow DApps to integrate and interact with the Appchain
+- **Storage** - this component manages the state storage of the Appchain in a highly efficient key-value database
+- **Consensus** - this component ensures that all the participants in the network agree on the state of the blockchain, validating transactions, state transitions, and the resulting blocks
+
+The default configuration of a Substrate node and the built-in implementations of the components are usually the best choice for most use cases. Still, teams are welcome to innovate and change or replace any piece of the node or even write a completely different implementation of the core client, such as [Kagome](https://github.com/soramitsu/kagome#intro){target=_blank} (C++ implementation) and [Gossamer](https://github.com/ChainSafe/gossamer#a-go-implementation-of-the-polkadot-host){target=_blank} (Golang implementation).
+
+## The Runtime {: #runtime }
```

Comment:
I like how in the core client section you covered some of the main components. In this section, you talk about the advantages of wasm, but maybe it would be beneficial to also cover some of the main components of the runtime?

Reply:
I think there is also an opportunity here to briefly mention pallets and then link to the modularity page

Reply:
Runtime is a very atomic component, but I left a smooth path towards modularity. wdyt ?

Reply:
yeah looks good!